pandas/tests/formats/test_format.py

```diff
@@ -3925,6 +3925,10 @@ def test_rounding(self):
         result = formatter(0)
         self.assertEqual(result, u(' 0.000'))
 
+    def test_nan(self):
+        formatter = fmt.EngFormatter(accuracy=1, use_eng_prefix=True)
+        result = formatter(np.nan)
+        self.assertEqual(result, u('NaN'))
 
```

Comment:
The example in the issue had random initialization, I have changed it so that I can expect the results.

issue:
df = pd.DataFrame(data = np.random.randint(25, size=(10, 3)), columns = list('abc'))

testcase:
 df = pd.DataFrame({'a':[1.5, 10.3, 20.5], 'b':[50.3, 60.67, 70.12], 'c':[100.2, 101.33, 120.33]})